pybamm/spatial_methods/spectral_volumes.py

```diff
@@ -0,0 +1,617 @@
+import pybamm
+
+import numpy as np
+
+from scipy.sparse import (
+    diags,
+    eye,
+    kron,
+    csr_matrix,
+    lil_matrix,
+    coo_matrix,
+)
+
+
+class SpectralVolumes(pybamm.FiniteVolume):
+    """
+    A class which implements the steps specific to the spectral volumes
+    discretisation. It is implemented in such a way that it is very
+    similar to FiniteVolume; that comes at the cost that it is only
+    compatible with the SpectralVolumes1DSubMesh (which is a certain
+    subdivision of any 1D mesh, so it shouldn't be a problem).
+
+    For broadcast and mass_matrix, we follow the default behaviour from
+    SpatialMethod. For spatial_variable, preprocess_external_variables,
+    divergence, divergence_matrix, laplacian, integral,
+    definite_integral_matrix, indefinite_integral,
+    indefinite_integral_matrix, indefinite_integral_matrix_nodes,
+    indefinite_integral_matrix_edges, delta_function
+    we follow the behaviour from FiniteVolume. This is possible since
+    the node values are integral averages with Spectral Volumes, just
+    as with Finite Volumes. delta_function assigns the integral value
+    to a CV instead of a SV this way, but that doesn't matter too much.
+    Additional methods that are inherited by FiniteVolume which
+    technically are not suitable for Spectral Volumes are
+    boundary_value_or_flux, process_binary_operators, concatenation,
+    node_to_edge, edge_to_node and shift. While node_to_edge (as well as
+    boundary_value_or_flux and process_binary_operators)
+    could utilize the reconstruction approach of Spectral Volumes, the
+    inverse edge_to_node would still have to fall back to the Finite
+    Volumes behaviour. So these are simply inherited for consistency.
+    boundary_value_or_flux might not benefit from the reconstruction
+    approach at all, as it seems to only preprocess symbols.
+
+    Parameters
+    ----------
+    mesh : :class:`pybamm.Mesh`
+        Contains all the submeshes for discretisation
+
+    **Extends:"": :class:`pybamm.FiniteVolume`
+    """
+
+    def __init__(self, options=None, order=2):

```

Comment:
`mesh` is not a parameter here? Also, when it is made a parameter, there should be a check that it is the `SpectralVolume1DSubMesh`, raising an error if not

Reply:
Well, it wasn't a parameter for the "FiniteVolume" constructor. I will overwrite the "build"-method where the "mesh" parameter is present and implement the check there.